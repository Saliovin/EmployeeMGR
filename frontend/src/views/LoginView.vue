<script setup lang="ts">
import type { User } from "@/interfaces/user";
import router from "@/router";
import service from "@/services/auth";
import { ref } from "vue";

const error_message = ref("")

const loginUser = (user: User) => {
  service
    .login(user)
    .then((res) => {
      router.push("/");
    })
    .catch((error) => {
      console.log(error)
      error_message.value = "Login Unsuccessful"
    });
};

const onSubmitLogin = (e: Event) => {
  error_message.value = ""
  const formData = new FormData(e.target as HTMLFormElement);
  const user = Object.fromEntries(formData);
  // @ts-ignore
  loginUser(user);
};
</script>

<template>
  <div class="login">
    <div class="panel">
      <h2>App Title</h2>
      <div class="error" v-if="error_message">{{ error_message }}</div>
      <form @submit.prevent="onSubmitLogin">
        <div class="input-group">
          <label>Username</label>
          <input name="username" />
        </div>
        <div class="input-group">
          <label>Password</label>
          <input name="password" type="password" />
        </div>
        <button type="submit">Login</button>
      </form>
    </div>
  </div>
</template>

<style scoped>
.login {
  display: flex;
  flex-direction: column;
  place-items: center;
  margin: 0 auto;
  padding: 2rem;
  max-width: 60%;
  height: 100%;
  background-color: #fff;
}

.panel {
  min-width: 15rem;
}

.input-group {
  display: flex;
  flex-direction: column;
}

h2 {
  text-decoration: underline;
  text-align: center;
  margin-bottom: 1rem;
}

button {
  width: 100%;
  margin-top: 0.5rem;
  background-color: #89b0de;
}

button:hover {
  background-color: #6678ba;
}
</style>
