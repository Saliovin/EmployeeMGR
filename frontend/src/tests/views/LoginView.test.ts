import LoginView from '@/views/LoginView.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('login', () => {
    it('base', () => {
        const login = mount(LoginView)
        expect(login.element).toMatchSnapshot()
    })
})