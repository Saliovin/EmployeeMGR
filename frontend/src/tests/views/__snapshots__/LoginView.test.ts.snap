// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`login > base 1`] = `
<div
  class="login"
  data-v-45f5edd7=""
>
  <div
    class="panel"
    data-v-45f5edd7=""
  >
    <h2
      data-v-45f5edd7=""
    >
      App Title
    </h2>
    <!--v-if-->
    <form
      data-v-45f5edd7=""
    >
      <div
        class="input-group"
        data-v-45f5edd7=""
      >
        <label
          data-v-45f5edd7=""
        >
          Username
        </label>
        <input
          data-v-45f5edd7=""
          name="username"
        />
      </div>
      <div
        class="input-group"
        data-v-45f5edd7=""
      >
        <label
          data-v-45f5edd7=""
        >
          Password
        </label>
        <input
          data-v-45f5edd7=""
          name="password"
          type="password"
        />
      </div>
      <button
        data-v-45f5edd7=""
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
